### DISCUSSION TITLE
책 4.3.6 패스워드 암호화 질문이요

### DISCUSSION THREAD

**Discussion by southbell2** [fsoftwareengineer/todo-application#37] [질문과 답변] [answered]
패스워드 암호화쪽 코드를 추가하고 나니 회원가입 후 로그인이 안 되더군요
찾아보니까 UserService.java 에서 getByCredentials 메소드

```
    public UserEntity getByCredentials(final String email, final String password, final PasswordEncoder encoder) {
        final UserEntity originalUser = userRepository.findByEmail(email);
        if (originalUser != null && encoder.matches(password, originalUser.getPassword())) {
            return originalUser;
        }
        return null;
    }
```
에서 encoder.matches(rawPassword, encodedPassword) 로 matches 메소드의 첫번째 인자는 원본 패스워드 값 , 두번째 인자는 인코딩된 패스워드 값이 들어가야 합니다.
그런데 저는 originalUser.getPassword() 를 해도 원본 패스워드 값이 나옵니다. 즉 "12345" 이런식으로 나오는 것이죠.
실제로 책을 보면서 공부를 했을때도 데이터베이스에 패스워드를 인코딩 시켜서 저장하는 것 같지가 않습니다.
혹시 책의 어느 부분에서 또는 코드의 어느 부분에서 데이터베이스에 패스워드를 인코딩 시켜서 저장하나요?

**Comment by fsoftwareengineer** [comment, marked as answer]
https://github.com/fsoftwareengineer/todo-application/discussions/3

위의 링크를 참고 해 주시기 바랍니다.
